experimental/Schemes/Schemes.jl

```diff
@@ -0,0 +1,2587 @@
+  pullbackFromParent::AlgHom
+  pullbackFromRoot::AlgHom
+  u::U # The inverse of denom in R
+  name::String
+  loc_var_name::String
+  #inclusionInParent::AffSchMorphism
+  #inclusionInRoot::AffSchMorphism
+
+  
+  function SpecPrincipalOpen(orig::Union{Spec{S,T,U},SpecPrincipalOpen{S,T,U}}, denom::U; loc_var_name="t" ) where {S <: Ring, T<:MPolyRing, U<:MPolyElem}
+    x = new{S,T,U}() 
+    x.parent = orig
+    # TODO: Implement a plausibility check: Does denom belong to the coordinate ring of the root?
+    parent(denom) == ambient_ring(root(x)) || error( "denominator has the wrong ring as parent" )
+    x.denom = denom
+    x.loc_var_name=loc_var_name
+    return x 
+  end 
+
+  function SpecPrincipalOpen( k::S, R::T, I::MPolyIdeal{U}; loc_var_name="t" ) where {S <: Ring, T<:MPolyRing, U<:MPolyElem}
+    if k != base_ring(R) 
+      error( "basering not compatible with ambient ring" )
+    end
+    if base_ring(I) != R
+      error( "ideal does not live in the given ambient ring" )
+    end
+    X = new{S,T,U}(k,R,I,loc_var_name)
+    return X
+  end
+end
+
+# Copy constructor
+@Markdown.doc """
+    SpecPrincipalOpen( X::SpecPrincipalOpen )
+
+Returns a copy of X
+"""
+function SpecPrincipalOpen( X::SpecPrincipalOpen )
+  Y = SpecPrincipalOpen( X.k, X.R, X.I )
+  if isdefined( X, :denom )
+    Y.denom = X.denom # this is safe to copy. It's excempt from "Fernwirkung" (tested).
+  end
+  # The following two variables are being cached. So we probably don't want 
+  # to copy them, because the need to be build on the new instance and not 
+  # accidentally refer to the member-variables of the original instance. 
+  # We should talk about this. 
+#  if isdefined( X, :pullbackFromParent )
+#    Y.pullbackFromParent = X.pullbackFromParent 
+#  end
+#  if isdefined( X, :pullbackFromRoot )
+#    Y.pullbackFromRoot = X.pullbackFromRoot
+#  end
+  if isdefined( X, :u )
+    Y.u = X.u
+  end
+  return Y
+end
+
+Base.copy( X::SpecPrincipalOpen ) = SpecPrincipalOpen( X )
+
+@Markdown.doc """
+    Spec( D::SpecPrincipalOpen ) 
+    
+Returns an instance of Spec isomorphic to D. This uses 
+the Rabinowitsch version for the localizations and forgets 
+about the ancestry in the localization tree.
+"""
+function Spec( D::SpecPrincipalOpen ) 
+  return Spec( base_ring(D), ambient_ring(D), defining_ideal(D) )
+end
+###############################################################################
+# Getter functions
+
+@Markdown.doc """
+    localize(X::AffineScheme{S,T,U}, f::U; loc_var_name="t" ) where {S,T,U}
+
+Returns an instance D of SpecPrincipalOpen where D = X ∖ Z(f).
+The optional argument `loc_var_name` can be used to specify a name 
+for printing the additional variable used in Rabinowitsch's trick 
+for localizations.
+"""
+function localize(X::AffineScheme{S,T,U}, f::U; loc_var_name="t" ) where {S,T,U}
+  return SpecPrincipalOpen(X, f; loc_var_name )
+end
+
+@Markdown.doc """
+    function parent(S::AffineScheme)
```

Comment:
```suggestion
   parent(S::AffineScheme)
```
The `function` must also be removed in a bunch of other docstrings below.